### PR TITLE
GH-1085: spring-rabbit-junit -> spring-rabbit-test

### PR DESCRIPTION
**Pull request by artembilan**
Fixes: https://github.com/spring-projects/spring-amqp/issues/1085
Fixes: https://github.com/spring-projects/spring-amqp/issues/1084

* Make a `spring-rabbit-junit` as compile dependency for the
`spring-rabbit-test` for better end-user experience
* Also move `What's New` chapter to the top level as it is with other
projects

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -383,13 +383,11 @@ project('spring-rabbit-test') {
...
+		compile project(':spring-rabbit-junit')
 		compile "org.hamcrest:hamcrest-library:$hamcrestVersion"
 		compile "org.hamcrest:hamcrest-core:$hamcrestVersion"
-		compile ("org.mockito:mockito-core:$mockitoVersion") {
-			exclude group: 'org.hamcrest', module: 'hamcrest-core'
-		}
+		compile "org.mockito:mockito-core:$mockitoVersion"
 		testCompile project(':spring-rabbit').sourceSets.test.output
-		testCompile project(':spring-rabbit-junit')
 	}
```
Comment (garyrussell) [inline build.gradle]: We should add something to the What's New - people who are using TestNG will now have to exclude it.

Comment (artembilan) [reply build.gradle]: Well, not sure how does that affect.
A `junit` and `junit-jupiter-api` are *optional* in the `spring-rabbit-junit`, but we indeed can add a note in What's New.

Comment (garyrussell) [reply build.gradle]: Oh, of course; never mind.

**Comment by garyrussell** [comment]
Merged as 34e423f8eeae06a57390306dfe52cc0912599542 with minor polishing.

**Pull request closed without merging**